src/helpers/index.tsx

```diff
@@ -9,139 +9,166 @@ import { abi as BondDaiContract } from "../abi/bonds/DaiContract.json";
-    return addresses[networkID].RESERVES.FRAX;
-  }
-  if (bond === BONDS.eth) {
-    return addresses[networkID].RESERVES.ETH;
+    return reserveAddresses.FRAX;
+  } else {
+    // bond === BONDS.eth
+    return reserveAddresses.ETH;
   }
 }
 
-export function isBondLP(bond) {
+export function isBondLP(bond: string) {
   return bond.indexOf("_lp") >= 0;
 }
 
-export function lpURL(bond) {
+export function lpURL(bond: string) {
   if (bond === BONDS.ohm_dai)
     return "https://app.sushi.com/add/0x383518188c0c6d7730d91b2c03a03c837814a899/0x6b175474e89094c44da98b954eedeac495271d0f";
   if (bond === BONDS.ohm_frax)
     return "https://app.uniswap.org/#/add/v2/0x853d955acef822db058eb8505911ed77f175b99e/0x383518188c0c6d7730d91b2c03a03c837814a899";
 }
 
-export function bondName(bond) {
+export function bondName(bond: string) {
   if (bond === BONDS.dai) return "DAI";
   if (bond === BONDS.ohm_dai) return "OHM-DAI SLP";
   if (bond === BONDS.ohm_frax) return "OHM-FRAX LP";
   if (bond === BONDS.frax) return "FRAX";
   if (bond == BONDS.eth) return "wETH";
 }
 
-export function contractForBond({ bond, networkID, provider }) {
+// TS-REFACTOR-NOTE: it may be worthy to create a mapping in a constants file to abstract a lot of this logic out.
+export function contractForBond({
+  bond,
+  networkID,
+  provider,
+}: {
+  bond: string;
+  networkID: number;
+  provider: StaticJsonRpcProvider | JsonRpcSigner;
+}) {
+  const bondAddresses = addresses[networkID].BONDS as Nested;
+
   if (bond === BONDS.ohm_dai) {
-    return new ethers.Contract(addresses[networkID].BONDS.OHM_DAI, BondOhmDaiContract, provider);
+    return new ethers.Contract(bondAddresses.OHM_DAI, BondOhmDaiContract, provider) as OlympusBondDepository;
   }
   if (bond === BONDS.dai) {
-    return new ethers.Contract(addresses[networkID].BONDS.DAI, BondDaiContract, provider);
+    return new ethers.Contract(bondAddresses.DAI, BondDaiContract, provider) as OlympusBondDepository;
   }
   if (bond === BONDS.ohm_frax) {
-    return new ethers.Contract(addresses[networkID].BONDS.OHM_FRAX, BondOhmFraxContract, provider);
+    return new ethers.Contract(bondAddresses.OHM_FRAX, BondOhmFraxContract, provider) as OlympusBondDepository;
   }
   if (bond === BONDS.frax) {
-    return new ethers.Contract(addresses[networkID].BONDS.FRAX, FraxBondContract, provider);
-  }
-  if (bond === BONDS.eth) {
-    return new ethers.Contract(addresses[networkID].BONDS.ETH, EthBondContract, provider);
+    return new ethers.Contract(bondAddresses.FRAX, FraxBondContract, provider) as OlympusBondDepository;
+  } else {
+    // bond === BONDS.eth
+    return new ethers.Contract(bondAddresses.ETH, EthBondContract, provider) as OlympusBondDepository;
   }
 }
 
-export function contractForReserve({ bond, networkID, provider }) {
+export function contractForReserve({
+  bond,
+  networkID,
+  provider,
+}: {
+  bond: string;
+  networkID: number;
+  provider: StaticJsonRpcProvider | JsonRpcSigner;
```

Comment:
hhm. Why is there an OR here?

Reply:
https://github.com/OlympusDAO/olympus-frontend/blob/ts-refactor-clean/src/actions/Bond.actions.ts#L69-L70
if you search `provider: signer` in this file you will notice several instances where we pass a signer instead of provider, I didn't want to mess w/ the logic so I just kept it as is.